b_pull_Landsat_SRST_poi/src/check_if_fully_within_pr.R

```diff
@@ -0,0 +1,87 @@
+#' @title Check to see if point is completely contained within path-row
+#' 
+#' @description
+#' Using the output of the previous target `b_WRS_pathrow_poi`, and the output of 
+#' the target `b_ref_locations_poi`, add WRS pathrow information to the locations 
+#' file, remove buffered points that are not completely within the 
+#' path row geometry.
+#' 
+#' @param WRS_pathrows list of pathrows to iterate over, output of target `b_WRS_pathrow_poi`
+#' @param locations dataframe of locations, output of target `b_ref_locations_poi`
+#' @param yml contents of the reformatted yaml .csv file, output of target `b_yml_poi`
+#' 
+#' @returns tibble of locations that includes WRS2 pathrow if, when buffered, they 
+#' are fully contained by the WRS2 pathrow extent
+#' 
+#' @note
+#' This step will result in more rows than the locations file, because a single 
+#' location in space can fall into multiple pathrows.
+#' 
+#' 
+check_if_fully_within_pr <- function(WRS_pathrow, locations, yml) {
+  # make a directory of locations for use in python workflow
+  if (!dir.exists("b_pull_Landsat_SRST_poi/out/locations/")) {
+    dir.create("b_pull_Landsat_SRST_poi/out/locations/")
+  }
+  # get the WRS2 shapefile
+  WRS <- read_sf("b_pull_Landsat_SRST_poi/in/WRS2_descending.shp")
+  # make locations into a {sf} object
+  locs <- st_as_sf(locations, 
+                   coords = c("Longitude", "Latitude"), 
+                   crs = yml$location_crs)
+  # make sure that the locs are in WGS84 if they aren't already
+  if (yml$location_crs != "EPSG:4326") {
+    locs <- st_transform(locs, "EPSG:4326")
+  }
+  # map over each path-row, adding the pathrow to the site. Note, this will create
+  # a larger number of rows than the upstream file, because sites can be in more
+  # than one pathrow. 
+  # filter for one path-row
+  one_PR <- WRS %>% filter(PR == WRS_pathrow) 
+  # get the locs within the path-row
+  x <- locs[one_PR, ]
+  x <- x %>% 
+    mutate(WRS2_PR = WRS_pathrow) 
+  
+  # in order to apply a buffer in sf, we need to convert to UTM, otherwise 
+  # it's assumed to be decimal degrees
+  # we'll use the WRS to calculate the appropriate UTM to use here.
+  wrs <- st_transform(one_PR, crs = "EPSG:4326")
```

Comment:
is one_PR already in WGS84? (EPSG 4326) in the shapefile I just downloaded it already is in 4326

Reply:
It may be! This was a sanity check. I'll change to an if/then just for thoroughness.